modules/class-simplechart-template.php

```diff
@@ -39,26 +38,32 @@ public function render_shortcode( $attrs ){
 	// render the chart from template
 	public function render( $id ){
 
-		// only allow non-published charts if we're looking at a post preview
+		// only allow non-published charts unless we're looking at a post preview
 		if ( 'publish' !== get_post_status( $id ) && ! is_preview() ){
 			return '';
 		}
 
-		$loader_url = Simplechart::instance()->get_config( 'widget_loader_url' );
-
-		// support versions that predate loading by chart ID
-		$json_data = get_post_meta( $id, 'simplechart-data', true );
-		$image_fallback = wp_get_attachment_image_src( get_post_thumbnail_id( $id ), 'large' );
-		$template_format = file_get_contents( Simplechart::instance()->get_plugin_dir() . 'templates/template-partial.html' );
-
-		$template_html = sprintf( $template_format,
-			json_encode( json_decode( $json_data ), JSON_HEX_APOS ),
-			Simplechart::instance()->save->validate_template_fragment( $template_html ),
-			esc_url( $loader_url ),
-			( ! $image_fallback ? '' : esc_url( $image_fallback[0] ) )
-		);
-
-		return $template_html;
+		if ( 'simplechart' !== get_post_type( intval( $id ) ) ) {
+			return '';
+		}
+		?>
+			<div
+				id="simplechart-widget-<?php echo esc_attr( $id ); ?>"
```

Comment:
you could also use `absint()` here too (instead of `esc_attr()`) for a tiny performance boost